mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/DebitCardScreen.kt

```diff
@@ -0,0 +1,204 @@
+                    }
+                },
+                keyboardActions = KeyboardActions(onDone =
+                {
+                    focusManager.moveFocus(FocusDirection.Next)
+
+                }),
+                keyboardOptions = KeyboardOptions.Default.copy(
+                    keyboardType = KeyboardType.Number,
+                    imeAction = ImeAction.Done
+                ),
+                decorationBox = {
+                    Row(horizontalArrangement = Arrangement.Center) {
+                        repeat(2) { index ->
+                            CharView(
+                                index = index,
+                                text = month
+                            )
+                            Spacer(modifier = Modifier.width(8.dp))
+                        }
+                    }
+                }
+            )
+            Text(
+                text = "/",
+                fontSize = 30.sp,
+                modifier = Modifier.padding(horizontal = 10.dp)
+            )
+
+            BasicTextField(
+                modifier = Modifier
+                    .focusRequester(c)
+                    .focusProperties {
+                        next = a
+                    },
+                value = TextFieldValue(year, selection = TextRange(year.length)),
+                onValueChange = {
+                    year = it.text
+                },
+                keyboardOptions = KeyboardOptions.Default.copy(
+                    keyboardType = KeyboardType.Number,
+                    imeAction = ImeAction.Done
+                ),
+                keyboardActions = KeyboardActions(onDone =
+                {
+                    viewModel.verifyDebitCard(cardNumber, month, year)
+                }),
+                decorationBox = {
+                    Row(horizontalArrangement = Arrangement.Center) {
+                        repeat(4) { index ->
+                            CharView(
+                                index = index,
+                                text = year
+                            )
+                            Spacer(modifier = Modifier.width(8.dp))
+                        }
+                    }
+                }
+            )
+        }
+    }
+    when (debitCardUiState) {
+        is DebitCardUiState.Initials -> {
+            // do nothing
+        }
+
+        is DebitCardUiState.Verifying -> {
+            MfLoadingWheel(
+                contentDesc = Constants.PLEASE_WAIT,
+                backgroundColor = Color.White
+            )
+        }
+
+        is DebitCardUiState.Verified -> {
+            onDebitCardVerified((debitCardUiState as DebitCardUiState.Verified).otp)
+        }
+
+        is DebitCardUiState.VerificationFailed -> {
+            onDebitCardVerificationFailed((debitCardUiState as DebitCardUiState.VerificationFailed).errorMessage)
+        }
+
+        else -> {}
+    }
+}
+
+@Preview
+@Composable
+fun DebitCardScreenPreview() {
```

Comment:
This preview can't be generated. It will always give you an error, Please read this https://medium.com/androidbytesensei/write-testable-compose-function-a381330b7d15

Reply:
Generated